### PR TITLE
Video annotation support

### PR DESCRIPTION
**Pull request by Xlient**
This PR aims to add Video Annotation support ( Extracting the text from the video and formatting it to a webvtt  formated file );

### CODE REVIEW

File: BlazorApp1/Services/VideoAnnoatationService.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+                foreach (TextAnnotation annotation in orderedTextAnnotations)
+                {
+                    TimeSpan startTime = annotation.Segments[0].Segment.StartTimeOffset.ToTimeSpan();
+                    TimeSpan endTime = annotation.Segments[0].Segment.EndTimeOffset.ToTimeSpan();
+
+
+                    subFile.WriteLine(lineCount);
+                    subFile.WriteLine($"0{startTime.Hours}:0{startTime.Minutes}:0{startTime.Seconds}.{startTime.Milliseconds} --> 0{ endTime.Hours}:0{ endTime.Minutes}:0{ endTime.Seconds}.{ endTime.Milliseconds}");
+                    //TODO create translation Method  then call it here
```
Comment (Xlient) [inline BlazorApp1/Services/VideoAnnoatationService.cs:95]: @WanjiruMambo 🙌🏾 If your feeling ok about doing translations just pop the method here

File: BlazorApp1/Program.cs [hunk 2]
```diff
@@ -1,11 +1,11 @@
-using BlazorApp1.Data;
+using BlazorApp1.Services;
 
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
 builder.Services.AddRazorPages();
 builder.Services.AddServerSideBlazor();
-builder.Services.AddSingleton<WeatherForecastService>();
+builder.Services.AddSingleton<VideoAnnoatationService>();
```
Comment (Xlient) [inline BlazorApp1/Program.cs:8]: @WanjiruMambo @CKanja  If you create a Service ( kind of like a module that is reusable) you have to register it here like I did.

File: BlazorApp1/Services/VideoAnnoatationService.cs (hunk 1 above)
Comment (wmambo) [reply BlazorApp1/Services/VideoAnnoatationService.cs:95]: Does it mean that the language has to be detected first or is already detected by the time we are getting to the translation

Comment (Xlient) [reply BlazorApp1/Services/VideoAnnoatationService.cs:95]: According to what i read in the docs google automatically detects the language in the video. We can provide hints to google so it doesnt have to guess . Same for the translation API we dont have to tell it what language it is but just say what language we want it translated into.

Comment (Xlient) [reply BlazorApp1/Services/VideoAnnoatationService.cs:95]: @WanjiruMambo I am going to merge this branch so you can have the latest code to work with

**Comment by wmambo** [comment]
public String TranslateLanguage()
{
  TranslationServiceClient client = TranslationServiceClient.Create();
  TranslateTextRequest request = new TranslateTextRequest
  {
    Contents = { "It is raining." }, //how to go about this section
    TargetLanguageCode = "fr-FR",
    Parent = new ProjectName(projectId).ToString()
     };
    TranslateTextResponse response = client.TranslateText(request);
    // response.Translations will have one entry, because request.Contents has one entry.
    Translation translation = response.Translations[0];
    Console.WriteLine($"Detected language: {translation.DetectedLanguageCode}");
    Console.WriteLine($"Translated text: {translation.TranslatedText}");
}

**Pull request merged**